packages/core/src/tracing/instrumentation/protocols/grpcJs.js

```diff
@@ -0,0 +1,504 @@
+
+        if (!isTracing) {
+          return origFn.apply(this, arguments);
+        }
+
+        const originalArgs = copyArgs(arguments);
+
+        if (isSuppressed) {
+          modifyArgs(name, originalArgs, null);
+          return origFn.apply(this, originalArgs);
+        }
+
+        /**
+         * https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/client.ts#L281
+         */
+        return instrumentedClientMethod(
+          this,
+          origFn,
+          originalArgs,
+          { host, port: port.toString() },
+          method,
+          requestStream,
+          responseStream,
+          (args, span) => {
+            return modifyArgs(name, args, span);
+          }
+        );
+      };
+    });
+  });
+}
+
+// See https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/client.ts
+function modifyArgs(name, originalArgs, span) {
+  const mockCallback = (newArgs, originalCb) => {
+    newArgs[newArgs.length - 1] = cls.ns.bind(function (err) {
+      span.d = Date.now() - span.ts;
+
+      if (err) {
+        const errorMessage = err.details || err.message;
+
+        if (errorMessage !== 'Cancelled') {
+          span.ec = 1;
+          if (errorMessage) {
+            span.data.rpc.error = errorMessage;
+          }
+        }
+      }
+
+      span.transmit();
+      originalCb.apply(this, arguments);
+    });
+  };
+
+  const setInstanaHeaders = metadata => {
+    if (metadata && metadata.set) {
+      metadata.set(constants.spanIdHeaderName, span.s);
+      metadata.set(constants.traceIdHeaderName, span.t);
+      metadata.set(constants.traceLevelHeaderName, '1');
+    }
+  };
+
+  const checkMetadataOptionsAndCallback = (method, serialize, deserialize, argument, metadata, options, callback) => {
+    const arg1 = metadata;
+    const arg2 = options;
+    const arg3 = callback;
+
+    let newMetadata;
+    let newOptions;
+    let newCallback;
+
+    if (typeof arg1 === 'function') {
+      newMetadata = new Metadata();
+      newOptions = {};
+      newCallback = arg1;
+
+      originalArgs[originalArgs.length - 1] = newMetadata;
+      originalArgs.push(newOptions);
+      originalArgs.push(newCallback);
+    } else if (typeof arg2 === 'function') {
+      if (arg1 instanceof Metadata) {
+        newMetadata = arg1;
+        newOptions = {};
+        newCallback = arg2;
+
+        originalArgs[originalArgs.length - 2] = newMetadata;
+        originalArgs[originalArgs.length - 1] = newOptions;
+        originalArgs.push(newCallback);
+      } else {
+        newMetadata = new Metadata();
+        newOptions = arg1;
+        newCallback = arg2;
+
+        originalArgs[originalArgs.length - 2] = newMetadata;
+        originalArgs[originalArgs.length - 1] = newOptions;
+        originalArgs.push(newCallback);
+      }
+    } else {
+      if (!(arg1 instanceof Metadata && arg2 instanceof Object && typeof arg3 === 'function')) {
+        throw new Error('Incorrect arguments passed');
```

Comment:
An instrumentation must never cause an error. This looks like we have detected that our assumptions about number/types of the incoming arguments were not correct. Which can totally happen: Even if the assumptions are correct for the current release, if grpc-js releases a new version that passes arguments differently, we are in trouble.

The correct reaction to noticing that our assumptions were not correct would be to make do with what we have. Examples: We don't have a metadata object? Well then we cannot inject headers. We do not are missing an argument that we usually inspect for a span annotation? Well, then we will not add that span annotation. We couldn't find the result callback? Well then we cannot wrap it. And finally, if we think that we cannot record any reasonably correct span at all (or we already know that we have no means to end the span later), we would need to simply stop tracing, call the original function with its original arguments (like we do when tracing is not active) and call it a day.